lib/main/atom/views/outline/semanticView.ts

```diff
@@ -0,0 +1,64 @@
+import {NavigationTreeComponent} from "./navigationTreeComponent"
+import {NavigationTreeViewModel} from "./semanticViewModel"
+import {initialize} from "./semanticViewPane"
+
+export const SEMANTIC_VIEW_URI = "atomts-semantic-view"
+
+export interface SemanticViewOptions {
+  navTree?: NavigationTreeViewModel | null
+}
+
+export interface SemanticViewSerializationData {
+  data: {navTree: NavigationTreeViewModel | null}
+  deserializer: "atomts-semantic-view/SemanticView"
+}
+
+export function deserializeSemanticView(serialized: SemanticViewSerializationData) {
+  // console.log('deserializeSemanticView -> ', serialized)// DEBUG
+  const view = new SemanticView(serialized.data)
+  initialize(view)
+  return view
+}
+
+export class SemanticView {
+  private comp: NavigationTreeComponent
+
+  public get element() {
+    return this.comp.element
+  }
+
+  constructor(public config: SemanticViewOptions) {
+    this.comp = new NavigationTreeComponent({navTree: (config && config.navTree) || null})
```

Comment:
:warning: `config && ` is pointless here, since `config` is always defined.

Reply:
yes, I will change it

for explanation: that's just my paranoid nature -- there could still be `null` value, if someone really wanted to pass it in, e.g. using `null as any`, but I guess in that case it really should fail...

Reply:
I would argue that while there can indeed be anything anywhere since Javascript is weakly typed, we really should strive to catch that stuff before runtime, and thankfully TS makes it relatively simple. Otherwise, our code will become an unsightly mess of unnecessary checks everywhere, and that's just sad.